pkg/dao/snapshots.go

```diff
@@ -120,8 +120,78 @@ func (sDao *snapshotDaoImpl) List(
 	return api.SnapshotCollectionResponse{Data: resp}, totalSnaps, nil
 }
 
+func (sDao *snapshotDaoImpl) ListByTemplate(
+	ctx context.Context,
+	orgID string,
+	template api.TemplateResponse,
+	repositorySearch string,
+	paginationData api.PaginationData,
+) (api.SnapshotCollectionResponse, int64, error) {
+	var snaps []api.SnapshotResponse
+	var totalSnaps int64
+	pulpContentPath, err := sDao.pulpClient.GetContentPath(ctx)
+	if err != nil {
+		return api.SnapshotCollectionResponse{}, 0, err
+	}
+
+	// Get snapshots for template date
+	var date api.Date
+	if template.UseLatest {
+		date = api.Date(time.Now())
+	} else {
+		date = api.Date(template.Date)
+	}
+	snapshotsForTemplateDate, err := sDao.FetchSnapshotsByDateAndRepository(ctx, orgID, api.ListSnapshotByDateRequest{
+		RepositoryUUIDS: template.RepositoryUUIDS,
+		Date:            date,
+	})
+	if err != nil {
+		return api.SnapshotCollectionResponse{}, totalSnaps, err
+	}
+
+	// Repository search/filter and ordering
+	sortMap := map[string]string{
+		"repository_name": "repo_name",
+		"created_at":      "snapshots.created_at",
+	}
+	order := convertSortByToSQL(paginationData.SortBy, sortMap, "repo_name ASC")
+	var filteredSnaps []models.Snapshot
+	query := readableSnapshots(sDao.db.WithContext(ctx), orgID).
+		Select("snapshots.*, STRING_AGG(repository_configurations.name, '') as repo_name").
+		Where("repository_configuration_uuid IN ?", template.RepositoryUUIDS).
+		Where(fmt.Sprintf("repository_configurations.name ILIKE '%%%s%%'", repositorySearch)).
```

Comment:
```suggestion
		Where("repository_configurations.name ILIKE ?'", fmt.Sprintf("%%%s%%", repositorySearch)).
```
I'd change this line a bit to use the question mark syntax. The question mark syntax means the query is parameterized, which helps prevent SQL injection.

Reply:
good catch 👍🏼 
I had trouble doing it like this: `Where("repository_configurations.name ILIKE '%?%'", repositorySearch)`, so I decided to use the `fmt.Sprintf()` approach, but in the moment I didn't think of this 😄
resolved [here](https://github.com/content-services/content-sources-backend/pull/796/commits/b93e539c29cf17c4e5adc0233d36a4a86ef66cc2) 🫡